osu.Game.Rulesets.Osu/Difficulty/OsuDifficultyCalculator.cs

```diff
@@ -23,6 +24,65 @@ public OsuDifficultyCalculator(Ruleset ruleset, WorkingBeatmap beatmap)
         {
         }
 
+        protected override double strainStep() { return section_length; }
+
+        //Throws list of difficulty values at sections. First value is possesed by two sections.
+        protected override List<double> DifficultySectionRating (IBeatmap beatmap, double timeRate)
+        {
+            if (!beatmap.HitObjects.Any())
+                return new List<double>();
+
+            OsuDifficultyBeatmap difficultyBeatmap = new OsuDifficultyBeatmap(beatmap.HitObjects.Cast<OsuHitObject>().ToList(), timeRate);
+            Skill[] skills =
+            {
+                new Aim(),
+                new Speed()
+            };
+
+            double sectionLength = section_length * timeRate;
+
+            // The first object doesn't generate a strain, so we begin with an incremented section end
+            double currentSectionEnd = 2 * sectionLength;
+
+            foreach ( OsuDifficultyHitObject h in difficultyBeatmap )
+            {
+                while (h.BaseObject.StartTime > currentSectionEnd)
+                {
+                    foreach (Skill s in skills)
```

Comment:
Prefer usage of `var`

Reply:
you mean `foreach (var s in skills)` or making it for loop and then var?

Reply:
The first one

Reply:
ok thanks